libkv/subscription.go

```diff
@@ -65,6 +65,50 @@ func (service Service) CreateSubscription(s *subscription.Subscription) (*subscr
 	return s, nil
 }
 
+// UpdateSubscription updates subscription.
+func (service Service) UpdateSubscription(id subscription.ID, s *subscription.Subscription) (*subscription.Subscription, error) {
+	err := service.validateSubscription(s)
+	if err != nil {
+		return nil, err
+	}
+
+    // If the subscriptionID changes, it should be a new subscription rather than an update.
```

Comment:
I think it should be approached differently. `validateSubscription` should check if `event`, `functionId`, `path` and `method` are the same, if not it should return validation error. If those values are the same we can just update the key in etcd. 

This approach causes that in some cases if we will change the subscription ID we will also need to change this logic. There is no need to do that.

Reply:
> This approach causes that in some cases if we will change the subscription ID we will also need to change this logic

Wouldn't we need a migration of existing subscriptions anyway in that case? Also, if we do change the how the subscriptionID is calculated, wouldn't that be implemented in the `generateSubscriptionID` method?

Reply:
It would, but we should not treat ID as something that contains meaningful ingo. It can be easily achieved by comparing current values with new values, it's just safer and more obvious. It's a bit more confusing the IDs are compared..